### PR TITLE
Integrations page

### PR DESCRIPTION
**Pull request by erika-cardenas**
Feature branch to gather changes for an integrations section

### Type of change:
- [x] **Documentation** updates (non-breaking change to fix/update documentation)

### How Has This Been Tested?
- [x] **Github action** – automated build completed without errors
- [x] **Local build** - the site works as expected when running `yarn start`

### CODE REVIEW

**Comment by daveatweaviate** [comment]
Hi @erika-cardenas is this a feature branch to collect changes for integration docs?

**Comment by daveatweaviate** [comment]
[staging](https://integrations-page--tangerine-buttercream-20c32f.netlify.app/developers/integrations)

**Review by daveatweaviate — COMMENTED** [review]
Hi @erika-cardenas  a couple of thoughts

- I added a link to [staging](https://integrations-page--tangerine-buttercream-20c32f.netlify.app/developers/integrations/data-pipeline) in the comments
- The branch needs to be refreshed. It's easier to do that locally. I'm not sure if you have other changes coming so I haven't refreshed it.
- The Data Platforms page shows up in the ToC twice. I think this page is in the wrong directory: /developers/integrations/data-platforms/confluent-cloud
- These pages are missing content. Consider adding an underscore to the file names to hide them if you need to publish the rest before you have content.

**Comment by erika-cardenas** [comment]
Hey @daveatweaviate, 
I'm merging the changes from the smaller branches now!

File: developers/integrations/operations/ragas/index.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+title: Ragas
+sidebar_position: 4
+image: og/integrations/home.jpg
+---
+[Ragas](https://docs.ragas.io/en/stable/) is a framework to help you evaluate your retrieval augmented generation (RAG) applications.
+
+## Ragas and Weaviate 
+You'll need to store the `question`, `answer`, `ground_truths`, and `contexts` in a json to load it into Ragas. The Ragas metrics are:
```
Comment (daveatweaviate) [inline developers/integrations/operations/ragas/index.md:9]: Consider
The Ragas metrics are:

File: developers/integrations/operations/ragas/index.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+* faithfulness
+* answer_relevancy
+* context_precision
+* context_recall
+
```
Comment (daveatweaviate) [inline developers/integrations/operations/ragas/index.md:14]: To use Ragas, convert your `question`, `answer`, `ground_truths`, and `contexts` to json, and then send them to Ragas.

File: developers/integrations/operations/wandb/index.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+---
+title: Weights and Biases
+sidebar_position: 5
+image: og/integrations/home.jpg
+---
+
+[Weights and Biases(W&B)](https://wandb.ai/site) is an AI developer platform with products such as `Models` and `Weave`. Weights and Biases is famous for `Models` and machine learning model training visualizations such as `runs` and `sweeps`.
```
Comment (daveatweaviate) [inline developers/integrations/operations/wandb/index.md:7]: Consider

[Weights and Biases(W&B)](https://wandb.ai/site) is an AI developer platform with products such as `Models` and `Weave`.

Weights and Biases is famous for `Models` and for machine learning model training visualizations such as `runs` and `sweeps`.

File: developers/integrations/operations/wandb/index.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+## Weights and Biases and Weaviate 
+`Weave` from W&B offers a user interface and data analysis APIs for monitoring applications using LLM, RAG, Agents, and Vector Databases.
```
Comment (daveatweaviate) [inline developers/integrations/operations/wandb/index.md:10]: Consider
`Weave` has a user interface and data analysis APIs for monitoring applications that use LLM, RAG, Agents, and Vector Databases.

**Review by daveatweaviate — COMMENTED** [review]
@erika-cardenas Here are some comments on the operations sections.

I'll send LLM frameworks next

File: developers/integrations/llm-frameworks/index.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+---
+title: LLM Frameworks
+sidebar_position: 4
+image: og/integrations/home.jpg
+---
+
+Large Language Model (LLM) frameworks contain several tools to help developers build applications with generative models.
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/index.md:7]: Consider
These Large Language Model (LLM) frameworks have tools to help developers build applications with generative models.

File: developers/integrations/llm-frameworks/index.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+
+Learn about how Weaviate integrates with these solutions:
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/index.md:10]: Consider
Learn how Weaviate

File: developers/integrations/llm-frameworks/composio/index.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
+---
+title: Composio
+sidebar_position: 1
+image: og/integrations/home.jpg
+---
+
+[Composio](https://docs.composio.dev/introduction/intro/overview) uses function calling for managing and integrating tools with language models and AI agents.
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/composio/index.md:7]: Consider
[Composio](https://docs.composio.dev/introduction/intro/overview) uses functions to manage language models and AI agents.

File: developers/integrations/llm-frameworks/composio/index.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
+---
+title: Composio
+sidebar_position: 1
+image: og/integrations/home.jpg
+---
+
+[Composio](https://docs.composio.dev/introduction/intro/overview) uses function calling for managing and integrating tools with language models and AI agents.  
+
+## Composio and Weaviate 
+With Weaviate's retrieval, you can make the agent more personalized and context-aware.
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/composio/index.md:10]: Consider
Personalize agents and make them more context aware.

File: developers/integrations/llm-frameworks/composio/index.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+sidebar_position: 1
+image: og/integrations/home.jpg
+---
+
+[Composio](https://docs.composio.dev/introduction/intro/overview) uses function calling for managing and integrating tools with language models and AI agents.  
+
+## Composio and Weaviate 
+With Weaviate's retrieval, you can make the agent more personalized and context-aware. 
+
+The integration is supported through our LangChain vector store. You need to have a running Weaviate instance and create the vector store with:
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/composio/index.md:12]: Consider

The integration is supported through our LangChain vector store. 

To configure the integration, create the vector store in your Weaviate instance:

File: developers/integrations/llm-frameworks/composio/index.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+## Composio and Weaviate 
+With Weaviate's retrieval, you can make the agent more personalized and context-aware. 
+
+The integration is supported through our LangChain vector store. You need to have a running Weaviate instance and create the vector store with: 
+```python
+WeaviateVectorStore.from_documents( )
+```
+
+You can learn more about how to create a vector store [here](https://python.langchain.com/v0.2/docs/integrations/vectorstores/weaviate/#step-1-data-import).
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/composio/index.md:17]: Consider 

Learn how to [create a vector store]

File: developers/integrations/llm-frameworks/dspy/dspy.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
+---
+title: DSPy
+sidebar_position: 1
+image: og/docs/more-resources.jpg
+---
+[DSPy](https://github.com/stanfordnlp/dspy) is a framework for programming language models created by Stanford NLP. DSPy introduces two key concepts: **programming model** and **optimizers**.
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/dspy/dspy.md:6]: Consider
[DSPy](https://github.com/stanfordnlp/dspy) is from Stanford NLP. It is a framework for programming language models.

DSPy introduces two key concepts, the **programming model** and **optimizers**.

File: developers/integrations/llm-frameworks/dspy/dspy.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+
+- **Programing model**: The programming model lets you define a series of components that make a language model request. Components include input and output fields, task descriptions, and calls to a vector database like Weaviate.
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/dspy/dspy.md:8]: Consider
Components include input fields, output fields, task descriptions, and calls to a vector database like Weaviate.

File: developers/integrations/llm-frameworks/dspy/dspy.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+
+- **Optimizers**: Optimizers compile your DSPy program to tune the language model prompt and/or the weights.
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/dspy/dspy.md:10]: Consider
model prompt and the model weights.

File: developers/integrations/llm-frameworks/dspy/dspy.md [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+
+## DSPy and Weaviate
+
+Weaviate is integrated with DSPy through the retriever module! You only need to connect to your Weaviate cluster through [WCD](https://console.weaviate.cloud/) or localhost, and pass in your collection:
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/dspy/dspy.md:14]: Consider
Weaviate is already integrated with DSPy!

To connect your Weaviate cluster [WCD](https://console.weaviate.cloud/) or local instance to DSPy, use the [retriever module](ADD LINK) and pass in your collection:

File: developers/integrations/llm-frameworks/dspy/dspy.md [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+
+```python
+weaviate_client = weaviate.Client("http://localhost:8080") # or pass in your WCD cluster url
+
+retriever_module = WeaviateRM("WeaviateBlogChunk", # collection name
+                    weaviate_client=weaviate_client)
+```
+
+## Our Resources 
+Here are a few resources on using DSPy from the Weaviate team!
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/dspy/dspy.md:24]: Consider
Here are a few resources from the Weaviate team on using DSPy!

File: developers/integrations/llm-frameworks/dspy/dspy.md [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+1. [**Hands on Learning**](#hands-on-learning): Build your technical understanding with end-to-end tutorials.
+
+2. [**Read and Listen**](#read-and-listen): Develop your conceptual understanding of these technologies.
+
+
+### Hands on Learning 
+
+| Topic | Description | Resource | 
+| --- | --- | --- |
+| Getting Started with RAG in DSPy | Learn about the four components needed to build a DSPy program: Installation, settings, datasets with `dspy.Example`, LLM metrics, DSPy programming model, and optimization. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/dspy/1.Getting-Started-with-RAG-in-DSPy.ipynb), [Video](https://youtu.be/CEuUG4Umfxs?si=4Gp8gR9glmoMJNaU) |
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/dspy/dspy.md:36]: There are more than four things listed. Consider

Learn about DSPy and how to build a program: Installation, settings, datasets, LLM metrics, DSPy programming model, and optimization.

File: developers/integrations/llm-frameworks/dspy/dspy.md [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+| BigQuery and Weaviate orchestrated with DSPy | Build an end-to-end RAG pipeline that uses BigQuery and Weaviate using DSPy. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/cloud-hyperscalers/google/bigquery/BigQuery-Weaviate-DSPy-RAG.ipynb)|
+
+
+
+### Read and Listen
+
+| Topic | Description | Resource | 
+| --- | --- | --- |
+| DSPy and ColBERT with Omar Khattab! - Weaviate Podcast #85 | Omar Khattab joins the Weaviate podcast to discuss DSPy and ColBERT. | [Video](https://www.youtube.com/watch?v=CDung1LnLbY) |
+| DSPy Explained| The core concepts of DSPy and walks through the introduction notebooks showing how to compile a simple retrieve-then-read RAG program and Multi-Hop RAG Program. | [Video](https://youtu.be/41EfOY0Ldkc?si=sFieUeHc9rXRn6uk)|
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/dspy/dspy.md:54]: Consider
The core concepts of DSPy. Walk through the introduction notebooks to compile a simple retrieve-then-read RAG program and Multi-Hop RAG Program.

File: developers/integrations/llm-frameworks/haystack/index.md [hunk 18]
```diff
@@ -0,0 +1,23 @@
+---
+title: Haystack
+sidebar_position: 3
+---
+
+[Haystack](https://haystack.deepset.ai/) is a language model framework for build large language model applications
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/haystack/index.md:6]: Typo and consider

[Haystack](https://haystack.deepset.ai/) is a framework for building large language model applications

File: developers/integrations/llm-frameworks/langchain/langchain.md [hunk 19]
```diff
@@ -0,0 +1,37 @@
+---
+title: LangChain
+sidebar_position: 4
+---
+
+[LangChain](https://python.langchain.com/v0.2/docs/introduction/) is a framework used for building applications around large language models (LLMs).
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/langchain/langchain.md:6]: Consider
is a framework for building applications that use large language models (LLMs).

File: developers/integrations/llm-frameworks/langchain/langchain.md [hunk 20]
```diff
@@ -0,0 +1,37 @@
...
+
+## LangChain and Weaviate 
+Weaviate is a supported [vector store](https://python.langchain.com/v0.2/docs/integrations/vectorstores/weaviate/#step-1-data-import) in LangChain. You will need a running Weaviate cluster (any deployment) to use the integration.
+
+You can connect to your Weaviate cluster:
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/langchain/langchain.md:11]: Consider

Connect LangChain to your Weaviate cluster:

File: developers/integrations/llm-frameworks/langchain/langchain.md [hunk 21]
```diff
@@ -0,0 +1,37 @@
...
+The resources are broken into two categories: 
+1. [**Hands on Learning**](#hands-on-learning): Build your technical understanding with end-to-end tutorials.
+
+2. [**Read and Listen**](#read-and-listen): Develop your conceptual understanding of these technologies.
+
+### Hands on Learning
+
+| Topic | Description | Resource | 
+| --- | --- | --- |
+| LangChain LCEL | This repository currently contains a notebook showcasing how to define a language program with LangChain LCEL, compile it with DSPy, and convert it back to LangChain LCEL. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/langchain/LCEL/RAG-with-LangChain-LCEL-and-DSPy.ipynb) |
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/langchain/langchain.md:27]: Consider
A notebook that defines a language program with LangChain LCEL, compiles it with DSPy, and converts it back to LangChain LCEL.

File: developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md [hunk 22]
```diff
@@ -0,0 +1,44 @@
+---
+title: LlamaIndex
+sidebar_position: 5
+image: og/docs/llamaindex.jpg
+---
+
+[LlamaIndex](https://www.llamaindex.ai/) is a framework for building large language model (LLM) applications. 
+
+## LlamaIndex and Weaviate
+Weaviate is a [supported vector store](https://docs.llamaindex.ai/en/stable/api_reference/storage/vector_store/weaviate/) in LlamaIndex. You need a running Weaviate instance to build the vector store. You can create it with:
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md:10]: Consider

Weaviate is a [supported vector store](https://docs.llamaindex.ai/en/stable/api_reference/storage/vector_store/weaviate/) in LlamaIndex.

Create a vector store for LlamaIndex:

File: developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md [hunk 23]
```diff
@@ -0,0 +1,44 @@
...
+The resources are broken into two categories: 
+1. [**Hands on Learning**](#hands-on-learning): Build your technical understanding with end-to-end tutorials.
+
+2. [**Read and Listen**](#read-and-listen): Develop your conceptual understanding of these technologies.
+
+### Hands on Learning
+
+| Topic | Description | Resource | 
+| --- | --- | --- |
+| Data Loaders in LlamaIndex | Learn how to load data into Weaviate using LlamaIndex, and connect LlamaIndex to an existing Weaviate cluster. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/llamaindex/data-loaders-episode1/episode1.ipynb) |
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md:26]: Learn how to load data into Weaviate using LlamaIndex, and how to connect LlamaIndex to an

File: developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md [hunk 24]
```diff
@@ -0,0 +1,44 @@
...
+| Simple Query Engine | Build a simple query engine | [Notebook](https://github.com/weaviate/recipes/tree/main/integrations/llm-frameworks/llamaindex/simple-query-engine) |
+| SQL Router Query Engine | Build a SQL Query Engine to search through your vector and SQL database. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/llamaindex/sql-router-query-engine/sql-query-router.ipynb) |
+| Sub-Question Query Engine | Build a query engine that will break down a complex question into multiple parts. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/llamaindex/sub-question-query-engine/sub_question_query_enginev1.ipynb) |
+| Advanced RAG | This notebook walks your through an advanced Retrieval-Augmented Generation (RAG) pipeline using LlamaIndex and Weaviate. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/llamaindex/retrieval-augmented-generation/advanced_rag.ipynb) |
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md:33]: Typo 
walks you

File: developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+| SQL Router Query Engine | Build a SQL Query Engine to search through your vector and SQL database. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/llamaindex/sql-router-query-engine/sql-query-router.ipynb) |
+| Sub-Question Query Engine | Build a query engine that will break down a complex question into multiple parts. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/llamaindex/sub-question-query-engine/sub_question_query_enginev1.ipynb) |
+| Advanced RAG | This notebook walks your through an advanced Retrieval-Augmented Generation (RAG) pipeline using LlamaIndex and Weaviate. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/llamaindex/retrieval-augmented-generation/advanced_rag.ipynb) | 
+| Naive RAG | This notebook walks your through a naive Retrieval-Augmented Generation (RAG) pipeline using LlamaIndex and Weaviate. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/llamaindex/retrieval-augmented-generation/naive_rag.ipynb)
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md:34]: Typo 
walks you

File: developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+
+
+
+### Read and Listen 
+| Topic | Description | Resource | 
+| --- | --- | --- |
+| Episode 1: Data Loading | This episode will show you how to connect your data to LlamaIndex and Weaviate.| [Video](https://youtu.be/Bu9skgCrJY8?feature=shared) |
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md:41]: Consider
This episode shows you how to

File: developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md [hunk 27]
```diff
@@ -0,0 +1,44 @@
...
+| Advanced RAG | This notebook walks your through an advanced Retrieval-Augmented Generation (RAG) pipeline using LlamaIndex and Weaviate. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/llamaindex/retrieval-augmented-generation/advanced_rag.ipynb) | 
+| Naive RAG | This notebook walks your through a naive Retrieval-Augmented Generation (RAG) pipeline using LlamaIndex and Weaviate. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/llamaindex/retrieval-augmented-generation/naive_rag.ipynb)
+
+
+
+### Read and Listen 
+| Topic | Description | Resource | 
+| --- | --- | --- |
+| Episode 1: Data Loading | This episode will show you how to connect your data to LlamaIndex and Weaviate.| [Video](https://youtu.be/Bu9skgCrJY8?feature=shared) | 
+| Indexes in LlamaIndex | We cover the three indexes (Vector Store Index, List Index, and Tree Index) and walk through the architecture design. The video ends with a demo using the Vector Store Index and List Index.  | [Video](https://youtu.be/6pLgOJrFL38?feature=shared) |
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md:42]: Consider
The video covers the three LlamaIndex indexes (Vector Store Index, List Index, and Tree Index) and walks through the architecture design.

File: developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md [hunk 28]
```diff
@@ -0,0 +1,44 @@
...
+| Episode 3: RAG Techniques in LlamaIndex | Learn about four query engines implemented in LlamaIndex. | [Video](https://youtu.be/Su-ROQMaiaw?feature=shared) |
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/llamaindex/LlamaIndex.md:43]: Consider
Learn about four query engines that are implemented with LlamaIndex.

File: developers/integrations/llm-frameworks/semantic-kernel/semantic-kernel.md [hunk 29]
```diff
@@ -0,0 +1,31 @@
+---
+title: Semantic Kernel
+sidebar_position: 6
+---
+Semantic Kernel is an LLM framework developed by Microsoft. Semantic Kernel facilitates building LLM applications with abstractions such as `plugins`, `memory`, `planners`, and more!
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/semantic-kernel/semantic-kernel.md:5]: Is there a link to the SK site?

File: developers/integrations/llm-frameworks/semantic-kernel/semantic-kernel.md [hunk 30]
```diff
@@ -0,0 +1,31 @@
...
+The resources are broken into two categories: 
+1. [**Hands on Learning**](#hands-on-learning): Build your technical understanding with end-to-end tutorials.
+
+2. [**Read and Listen**](#read-and-listen): Develop your conceptual understanding of these technologies.
+
+### Hands on Learning
+
+| Topic | Description | Resource | 
+| --- | --- | --- |
+|  RAG Chatbot with Semantic Kernel | Implements a simple workflow of retrieve then generate. using semantic kernel to help with prompt engineering and orchestrating OpenAI API calls and Weaviate as knowledgebase from which to retreive semantically relevant context. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/semantic-kernel/dotnet/Chatbot_RAG_Weaviate.ipynb) |
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/semantic-kernel/semantic-kernel.md:21]: Typo and consider
retrieve

Implements a simple workflow of retrieve-then-generate. Use a semantic kernel for prompt engineering, orchestrating OpenAI API calls, and integrating Weaviate. Use Weavate as a knowledge base to retrieve semantically relevant context.

File: developers/integrations/llm-frameworks/semantic-kernel/semantic-kernel.md [hunk 31]
```diff
@@ -0,0 +1,31 @@
...
+| Retrieval Augmented Generation using Weaviate and SK | Implements a simple workflow of retrieve then generate. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/semantic-kernel/RetrievalAugmentedGeneration_Weaviate.ipynb) |
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/semantic-kernel/semantic-kernel.md:22]: Consider

Implements a simple workflow of retrieve-then-generate.

File: developers/integrations/llm-frameworks/semantic-kernel/semantic-kernel.md [hunk 32]
```diff
@@ -0,0 +1,31 @@
...
+| Weasviate Persistent Memory | This notebook shows how to replace the `VolatileMemoryStore` memory storage with a `WeaviateMemoryStore` persistent memory. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/llm-frameworks/semantic-kernel/weaviate-persistent-memory.ipynb) |
```
Comment (daveatweaviate) [inline developers/integrations/llm-frameworks/semantic-kernel/semantic-kernel.md:23]: Typo and consider
Weaviate Persistent Memory

This notebook shows how to replace `VolatileMemoryStore` memory storage with a `WeaviateMemoryStore` persistent memory.

**Review by daveatweaviate — COMMENTED** [review]
Thanks @erika-cardenas 

Data platforms next (maybe that's not new?)

File: developers/integrations/data-platforms/firecrawl/index.md [hunk 33]
```diff
@@ -0,0 +1,19 @@
+---
+title: Firecrawl
+sidebar_position: 3
+image: og/integrations/home.jpg
+---
+
+[Firecrawl](https://www.firecrawl.dev/) is an AI first web scraping tool that allows you to easily crawl and extract clean, structured data from websites. It is an API service that converts a URL into clean markdown or structured data.
```
Comment (daveatweaviate) [inline developers/integrations/data-platforms/firecrawl/index.md:7]: Consider
is an AI-first web scraping tool

File: developers/integrations/data-platforms/firecrawl/index.md [hunk 34]
```diff
@@ -0,0 +1,19 @@
+---
+title: Firecrawl
+sidebar_position: 3
+image: og/integrations/home.jpg
+---
+
+[Firecrawl](https://www.firecrawl.dev/) is an AI first web scraping tool that allows you to easily crawl and extract clean, structured data from websites. It is an API service that converts a URL into clean markdown or structured data. 
+
+## Firecrawl and Weaviate 
+Firecrawl handles the complexities of web scraping like proxies, caching, rate limits, and dynamic content, delivering markdown or JSON output that is ready to be ingested into vector databases like Weaviate.
```
Comment (daveatweaviate) [inline developers/integrations/data-platforms/firecrawl/index.md:10]: Consider
Firecrawl handles the complexities of web scraping like proxies, caching, rate limits, and dynamic content. It produces markdown or JSON formatted output that is ready to be ingested into vector databases like Weaviate.

File: developers/integrations/data-platforms/firecrawl/index.md [hunk 35]
```diff
@@ -0,0 +1,19 @@
...
+
+## Our Resources 
+[**Hands on Learning**](#hands-on-learning): Build your technical understanding with end-to-end tutorials.
+
+### Hands on Learning
+
+| Topic | Description | Resource | 
+| --- | --- | --- |
+| Firecrawl to Weaviate | This notebook will show you how to scrape webpages using Firecrawl and load it into Weaviate. | [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/data-platforms/web-search/firecrawl/firecrawl-to-weaviate.ipynb)
```
Comment (daveatweaviate) [inline developers/integrations/data-platforms/firecrawl/index.md:19]: Consider

This notebook shows you how to scrape webpages with Firecrawl and load their contents into Weaviate.

File: developers/integrations/data-platforms/spark/index.md [hunk 36]
```diff
@@ -0,0 +1,31 @@
+---
+title: Spark
+sidebar_position: 4
+image: og/integrations/home.jpg
+---
+
```
Comment (daveatweaviate) [inline developers/integrations/data-platforms/spark/index.md:6]: Shoudl PySpark have a link too?

File: developers/integrations/data-platforms/spark/index.md [hunk 37]
```diff
@@ -0,0 +1,31 @@
...
+[Apache Spark](https://spark.apache.org/docs/latest/api/python/index.html) (or the Python API, PySpark) is an open-source data processing framework used for real-time, large-scale data processing. 
+
+## Spark and Weaviate
+The Spark connector enables you to easily ingest data from Spark data structures into Weaviate. 
+
+You can learn more about the Weaviate connector for Spark in [this repository](https://github.com/weaviate/spark-connector).
```
Comment (daveatweaviate) [inline developers/integrations/data-platforms/spark/index.md:12]: Consider

Learn more about the connector in the Weaviate [Spark connector repository](https://github.com/weaviate/spark-connector).

File: developers/integrations/data-platforms/unstructured/index.md [hunk 38]
```diff
@@ -0,0 +1,32 @@
+---
+title: Unstructured
+sidebar_position: 5
+image: og/integrations/home.jpg
+---
+
+[Unstructured](https://unstructured.io/) offers a platform and tools for ingesting and processing unstructured data for building retrieval augmented generation (RAG) applications.
```
Comment (daveatweaviate) [inline developers/integrations/data-platforms/unstructured/index.md:7]: Consider

[Unstructured](https://unstructured.io/) offers a platform and tools to work with unstructured data. Ingest and process unstructured data to use in retrieval augmented generation (RAG) applications.

**Review by daveatweaviate — COMMENTED** [review]
@erika-cardenas Here's the next one.

I think we already did the compute infrastructure and cloud hyperscalers. Let me know if there are any changes I should look at there.

File: developers/integrations/operations/index.md [hunk 39]
```diff
@@ -0,0 +1,17 @@
+---
+title: Operations
+sidebar_position: 5
+image: og/integrations/home.jpg
+---
+
+The following integrations with Weaviate facilitate operations associated with running Vector Database applications.
```
[the remark this answers is not in the document]
Comment (erika-cardenas) [reply developers/integrations/operations/index.md:7]: Combined the two sentences with this, "These operational integrations make it easier to monitor and evaluate AI Native applications."

File: developers/integrations/operations/nomic/index.md [hunk 40]
```diff
@@ -0,0 +1,20 @@
+---
+title: Nomic
+sidebar_position: 3
+image: og/integrations/home.jpg
+---
+[Nomic AI's] Atlas  is a visualization tool for vector embeddings. This describes reducing the dimensionality of high-dimensional vectors into 2 or 3 dimensions with algorithms such as t-SNE, UMAP, or PCA. These vectors can then be easily visualized by humans in scatter plots.
```
[the remark this answers is not in the document]
Comment (erika-cardenas) [reply developers/integrations/operations/nomic/index.md:6]: Yes 😅

**Comment by erika-cardenas** [comment]
Thank you so much @daveatweaviate! Just made the final edits. 🙂

**Comment by daveatweaviate** [comment]
@erika-cardenas thanks for all the updates, this is a big project!

I fixed the merge conflict and made a couple of tiny tweaks. 

This LGTM - you can merge whenever you want, or else I can merge it when we get back on Tuesday. Just me know.

File: developers/integrations/data-platforms/firecrawl/index.md [hunk 41]
```diff
@@ -0,0 +1,21 @@
...
+sidebar_position: 3
+image: og/integrations/home.jpg
+---
+
+[Firecrawl](https://www.firecrawl.dev/) is an AI-first web scraping tool.
+
+It allows you to easily crawl and extract clean, structured data from websites. It is an API service that converts a URL into clean markdown or structured data.
+
+## Firecrawl and Weaviate
+ Firecrawl handles the complexities of web scraping like proxies, caching, rate limits, and dynamic content. It produces markdown or JSON formatted output that is ready to be ingested into vector databases like Weaviate.
```
Comment (network-charles) [inline developers/integrations/data-platforms/firecrawl/index.md:12]:
```suggestion
Firecrawl handles the complexities of web scraping like proxies, caching, rate limits, and dynamic content. It produces markdown or JSON formatted output that is ready to be ingested into vector databases like Weaviate.
```

File: developers/integrations/data-platforms/spark/index.md [hunk 42]
```diff
@@ -0,0 +1,31 @@
...
+sidebar_position: 4
+image: og/integrations/home.jpg
+---
+
+[Apache Spark](https://spark.apache.org/docs/latest/api/python/index.html) (or the Python API, [PySpark](https://spark.apache.org/docs/latest/api/python/index.html#:~:text=PySpark%20is%20the%20Python%20API,for%20interactively%20analyzing%20your%20data.)) is an open-source data processing framework used for real-time, large-scale data processing. 
+
+## Spark and Weaviate
+The Spark connector enables you to easily ingest data from Spark data structures into Weaviate. 
+
+Learn more about the connector in the Weaviate[ Spark connector repository](https://github.com/weaviate/spark-connector).
```
Comment (network-charles) [inline developers/integrations/data-platforms/spark/index.md:12]:
```suggestion
Learn more about the connector in the Weaviate [Spark connector repository](https://github.com/weaviate/spark-connector).
```

File: developers/integrations/compute-infrastructure/modal/index.md [hunk 43]
```diff
@@ -0,0 +1,20 @@
+---
+title: Modal
+sidebar_position: 1
+---
+
+[Modal](https://modal.com/) provides a serverless platform that has on-demand access to GPUs and a custom high-performance container runtime. Use Modal to easily deploy and automatically scale high-performance applications.
+
+## Modal and Weaviate
+Weaviate leverages Modal's serverless infrastructure for fast embedding generation and for fast generative model calls.
```
Comment (network-charles) [inline developers/integrations/compute-infrastructure/modal/index.md:9]: The words "for fast" are repetitive.

```suggestion
Weaviate leverages Modal's serverless infrastructure for fast embedding generation and quick generative model calls.
```

File: developers/integrations/data-platforms/context-data/index.md [hunk 44]
```diff
@@ -0,0 +1,25 @@
...
+VectorETL by [Context Data](https://contextdata.ai/) is a modular no-code Python framework that is designed to help AI and Data Engineers work with data:
+
+* Quickly extract data from multiple data sources (databases, cloud storage, and local files)
+* Embed using major models (including OpenAI, Cohere, and Google Gemini)
+* Write to vector databases 
+
+## Context Data and Weaviate
+Weaviate is a [target connection](https://context-data.gitbook.io/context-data-1/adding-target-connections#add-a-weaviate-target-connection) in Context Data. 
+
+To connect to ContextData, open the console and input your Weaviate instance's URL and authentication credentials when prompted.
```
Comment (network-charles) [inline developers/integrations/data-platforms/context-data/index.md:16]:
```suggestion
To connect to Context Data, open the console and input your Weaviate instance's URL and authentication credentials when prompted.
```

**Comment by eddumelendez** [comment]
Hi, sorry for hijacking the PR. I'm really glad to see this coming as I raised a PR #1715 some time ago. It would be great to see the libraries listed there in this section. I can also contribute

**Comment by erika-cardenas** [comment]
Thank you @network-charles for looking this over! 

@eddumelendez Your excitement for this makes me super happy! I'm merging this PR into main soon; I hope you like the format and content.

**Comment by network-charles** [comment]
It’s a pleasure, @erika-cardenas.

**Comment by daveatweaviate** [comment]
@network-charles thank you for your suggestions!  

We'd like to give you credit for the suggestions. In order to include them, we need to verify that you have accepted our [Contributor license agreement](https://weaviate.io/service/contributor-license-agreement)

Would you open a new PR against our main branch for your suggestions? If you accept the Contributor license, please say so in the PR comments.

Thanks!

**Pull request merged**